src/Features/VisualBasic/Portable/CodeCleanup/VisualBasicCodeCleanupService.vb

```diff
@@ -0,0 +1,175 @@
+                    {IDEDiagnosticIds.UseObjectInitializerDiagnosticId, IDEDiagnosticIds.UseCollectionInitializerDiagnosticId}),
+                New DiagnosticSet(VBFeaturesResources.Apply_using_directive_placement_preferences,
+                    {IDEDiagnosticIds.MoveMisplacedUsingDirectivesDiagnosticId}),
+                New DiagnosticSet(VBFeaturesResources.Apply_file_header_preferences,
+                    {IDEDiagnosticIds.FileHeaderMismatch}))
+
+        Public Async Function CleanupAsync(currentDocument As Document,
+                                           enabledDiagnostics As EnabledDiagnosticOptions,
+                                           progressTracker As IProgressTracker,
+                                           cancellationToken As CancellationToken) As Task(Of Document) Implements ICodeCleanupService.CleanupAsync
+            ' add one item for the 'format' action we'll do last
+            If enabledDiagnostics.FormatDocument Then
+                progressTracker.AddItems(1)
+            End If
+
+            ' and one for 'remove/sort Imports' if we're going to run that.
+            Dim organizeUsings As Boolean = enabledDiagnostics.OrganizeUsings.IsRemoveUnusedImportEnabled OrElse
+                enabledDiagnostics.OrganizeUsings.IsSortImportsEnabled
+            If organizeUsings Then
+                progressTracker.AddItems(1)
+            End If
+
+            If _codeFixServiceOpt IsNot Nothing Then
+                currentDocument = Await ApplyCodeFixesAsync(currentDocument,
+                                                            enabledDiagnostics.Diagnostics,
+                                                            progressTracker,
+                                                            cancellationToken).ConfigureAwait(False)
+            End If
+
+            ' do the remove Imports after code fix, as code fix might remove some code which can results in unused Imports.
+            If organizeUsings Then
+                progressTracker.Description = VBFeaturesResources.Organize_Usings
+                currentDocument = Await RemoveSortUsingsAsync(currentDocument,
+                                                              enabledDiagnostics.OrganizeUsings,
+                                                              cancellationToken).ConfigureAwait(False)
+                progressTracker.ItemCompleted()
+            End If
+
+            If enabledDiagnostics.FormatDocument Then
+                progressTracker.Description = FeaturesResources.Formatting_document
+                Using Logger.LogBlock(FunctionId.CodeCleanup_Format, cancellationToken)
+                    currentDocument = Await Formatter.FormatAsync(currentDocument, cancellationToken:=cancellationToken).ConfigureAwait(False)
+                    progressTracker.ItemCompleted()
+                End Using
+            End If
+
+            Return currentDocument
```

Comment:
this logic looks like it would be identical to C#.  i would have expected there to be a shared base class extracted from the C# version, and the Vb and C# versions would subclass it and only od things specific to those languages.

Reply:
@CyrusNajmabadi  It was and worked perfectly in 2 previews (though no tests) and then someone disabled it.

Reply:
ok.  so to move forward.  we should move all the duplicated logic you have here to a single place (ideally an abstract base class).  VB and C# will then subclass that.